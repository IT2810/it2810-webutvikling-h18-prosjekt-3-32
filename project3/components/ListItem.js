import React from 'react';
import {AppState, AsyncStorage, Image, Modal, TouchableOpacity, Text, View, TextInput} from 'react-native';
import styles from "../stylesheets/ListItem.style.js";
import DatePicker from 'react-native-datepicker';
import { CheckBox } from 'react-native-elements'

export default class ListItem extends React.Component {
    constructor(props){
        super(props);
        this.state = ({
            date:"",
            name:"",
            modalVisible: false,
            dueDate: false,
            done:false,
            appState: AppState.currentState,
        });
    }

    //Since both date and name may be changed, the state is set by the props which comes from TodoList.js
    //When the name or date is changed, the state is changed and the user get immediate update!
    componentDidMount(){
        AppState.addEventListener('change', this.handleAppStateChange);
        this.setState({
            date: this.props.date,
            name: this.props.name,
            done: this.props.done,
        });
        this.checkDueDate();
    }

    componentWillUnmount(){
        AppState.removeEventListener('change', this.handleAppStateChange);
    }

    //This function checks the current state of the device (active, background, inactive)
    //If the app is closed, this.handleFinishedTodo will be triggered
    handleAppStateChange = (nextAppState) => {
        if (this.state.appState.match(/inactive|background/) && nextAppState === 'active') {

        }
        else{this.handleFinishedTodo()}
        this.setState({appState: nextAppState});
    };

    //This function only run when the application is closed.
    //It checks whether a to-do is marked as done or not, if it is (when the application is closed), the to-do is removed from the list
    //The to-do is also renamed in the AsyncStorage (see line 69 in TodoList.js for more!)
    handleFinishedTodo(){
        if(this.state.done){
            let finishedList = [this.props.todoNr, this.state.name, this.state.date, this.state.done];
            this.props.handleFinishedTodo(finishedList);
        }
    }

    //This Function checks whether a given due date to an to-do is today or have been
    //The text will be red, as an reminder for the user
    checkDueDate(){
        const today = new Date();
        const dueDate = new Date(this.props.date);
        if(dueDate <= today){
            this.setState({dueDate:true});
        }
        else{
            this.setState({dueDate:false})
        }
    }

    //Checks if the to-do is done or not, when the user press the checkbutton
    handlePressedCheckbox(){
        if(this.props.done){
            alert("This todo is already finished");
        }
        else{
            this.setState({done:!this.state.done});
        }
    }

    handleChangedDate(date){
        this.setState({
            date: date,
        });
    }

    //Set the modal visible, if the to-do isn't finished.
    setModalVisible(visible) {
        if(this.props.done){
            alert("This todo is already finished");
        }
        else{
            this.setState({modalVisible: visible});
        }
    }

    //deleteTodo delete an item with the given id.
    //this.props.deleteTodo is a function passed down to ListItem from TodoList.
    deleteTodo(id){
        this.props.deleteTodo(id);
    };

    //saveChanges is an async function which saves the changes to an to-do done by the user.
    //the new changes are stored in the Asyncstorage by the given todonr
    saveChanges = async () => {
        this.setModalVisible(!this.state.modalVisible);
        let list = [this.state.name, this.state.date];
        try {
            await AsyncStorage.setItem(this.props.todoNr.toString(), JSON.stringify(list));
        } catch (error) {
            throw error;
        }

        //Creates a list to update the list immediately when the user have changed the date or name
        const updateList = [this.props.todoNr, this.state.name, this.state.date];
        this.props.updateSortedList(updateList);
        this.checkDueDate();
    };

    render() {
        return (
            <React.Fragment>
                <TouchableOpacity onPress={() => {this.setModalVisible(true);}} style={this.state.done ? styles.topDone : styles.todoItemTop}>
                    <CheckBox
                        containerStyle={styles.checkBtn}
                        checked={this.state.done}
                        onIconPress={() => this.handlePressedCheckbox()}
                    />
                    <Text style={this.state.dueDate ? styles.itemRed : styles.itemText}>{this.state.name}</Text>
                    {/*Button to open the modal, where the user may enter date, save and delete the todo*/}
                    <TouchableOpacity
                        onPress={() => {
                            this.setModalVisible(true);
                        }}>
                        <Image style={styles.todoInfo} source={require('../assets/edit.png')}/>
                    </TouchableOpacity>
                </TouchableOpacity>
                <View style={this.state.done ? styles.bottomDone : styles.todoItemBottom}>
                    <Text style={styles.dateText}>{this.state.date}</Text>
                </View>
                <View>
                    {/*Modal which contains several options to the user regarding the todo*/}
                    <Modal
                        animationType="slide"
                        transparent={true}
                        visible={this.state.modalVisible}
                        onRequestClose={()=>{this.setModalVisible(!this.state.modalVisible);}}>
                        <View style={styles.modalView}>
                          <View style={styles.modal}>
                            {/*Simple backbutton if the user choose to not do any changes*/}
                            <View style={styles.newTodoTitleBar}>
                              <Text style={styles.newTodoTitle}>{"Todo info"}</Text>
                            </View>
                            <TouchableOpacity
                                onPress={() => {this.setModalVisible(!this.state.modalVisible);}}
                                style={styles.modalClose}>
                                <Image style={styles.closeImg} source={require('../assets/close.png')}/>
                            </TouchableOpacity>
                            {/*The user may change the todoText*/}
                            <TextInput style={styles.modalText}
                                       onChangeText={(text) => this.setState({name:text})}
                                       value={this.state.name}/>
                            <View style={styles.modalItem}>
                                {/*Date picker so the user could choose an due date to the todo.*/}
                                <DatePicker
                                    style={styles.datePicker}
                                    date={this.state.date}
                                    mode="date"
                                    placeholder="select date"
                                    minDate={new Date()}
                                    format="YYYY-MM-DD"
                                    confirmBtnText="Confirm"
                                    cancelBtnText="Cancel"
                                    onDateChange={(date) => {this.handleChangedDate(date)}}
                                />
                            </View>

                            <View style={styles.buttonsView}>
                                  {/*Button to save the changes done in the modal*/}
                                  <TouchableOpacity style={styles.saveButton} onPress={() => {this.saveChanges()}}>
                                      <Text style={styles.saveText}>Save</Text>
                                  </TouchableOpacity>
                                  {/*Button to delete the todo*/}
                                  <TouchableOpacity style={styles.deleteButton} onPress={() => this.deleteTodo(this.props.todoNr)}>
                                      <Text style={styles.deleteText}>Delete</Text>
                                  </TouchableOpacity>
                            </View>

                        </View>
                      </View>
                    </Modal>
                </View>
            </React.Fragment>
        );
    }
};