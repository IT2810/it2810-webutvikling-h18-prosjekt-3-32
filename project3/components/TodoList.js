import React from 'react';
import { Button, AsyncStorage, ScrollView, TextInput, TouchableOpacity, Text, View } from 'react-native';
import ListItem from "./ListItem.js";
import styles from "../stylesheets/TodoList.style.js";

export default class TodoList extends React.Component {
    constructor(props){
        super(props);

        //State with current todonr, which increases, and the current todotext that the user is entering
        this.state = {
            currTodoNr: 1,
            todoText: "",
            showFinishedTodos: true,
            todoList: [],
            finishedTodoList:[],
            showList:[],
        };

        //Binding functions
        this.updateTodoList = this.updateTodoList.bind(this);
        this.updateShowList = this.updateShowList.bind(this);
        this.addTodo = this.addTodo.bind(this);
        this.updateSortedList = this.updateSortedList.bind(this);
        this.handleFinishedTodo = this.handleFinishedTodo.bind(this);
        this.deleteTodo = this.deleteTodo.bind(this);
        this.myTextInput = React.createRef();
    };

    componentDidMount(){
        let todoList = [];
        let finishedTodoList = [];
        //Gets the highest todoNr and updates the state to no todos have the same todoNr and key.
        AsyncStorage.getAllKeys((err, keys) => {
            AsyncStorage.multiGet(keys, (err, stores) => {
                stores.map((result, i, store) => {
                    // get at each store's key/value so you can work with it
                    let key = store[i][0];
                    let value = store[i][1];
                    if(key === "CurrentTodoNr"){
                        this.setState({currTodoNr:parseInt(value)+1});
                    }

                    //Checks if the item is a to-do
                    if(key.substring(0,4) === "todo"){
                        const valueList = JSON.parse(value);
                        todoList.push({
                            key: key,
                            todoNr: key,
                            todoText: valueList[0],
                            todoDate: valueList[1],
                            done: valueList[2],
                        });
                    }

                    //Checks if the item is a finished to-do
                    else if (key.substring(0,4) === "done"){
                        const valueList = JSON.parse(value);
                        finishedTodoList.push({
                            key: key,
                            todoNr: key,
                            todoText: valueList[0],
                            done: valueList[2],
                        });
                    }
                });

                //Sorts the imported list from AsyncStorage and updates the state to all three lists
                const sortedByDateTodos = this.sortByDate(todoList);
                const sortedByDateFinishedTodos = this.sortByDate(finishedTodoList);
                this.setState({
                    showList:sortedByDateTodos,
                    todoList:sortedByDateTodos,
                    finishedTodoList:sortedByDateFinishedTodos,
                });
            });
        });
    }

    //addTodo adds a new to-do, add it to a list, and passes the list to app.js, since the to-do-list is needed at the home screen.
    //Each element in addTodo gets a key, todoNr, a name and a date(which is empty until the users sets a due date)
    addTodo = (id) => {
        this.myTextInput.current.clear();
        if(this.state.todoText !== ""){
            //Creates a list, sliced from the current todolist
            //Adds the new to-do to the list.
            const addList = this.state.todoList.slice();
            addList.push({
                key: id,
                todoNr: "todo"+id,
                todoText: this.state.todoText,
                todoDate: "",
                done: false,
            });
            this.setState({
                currTodoNr: this.state.currTodoNr + 1,
                todoText : "",
            });

            //The list is being sat to both to-do- and showlist for immediate change on the screen.
            this.updateTodoList(addList);
            this.updateShowList(addList);

            //The new to-do is also added in the AsyncStorage, aswell as the latest todoNr.
            const AsyncList = [this.state.todoText, "", false];
            this.storeTodo("todo"+id.toString(), JSON.stringify(AsyncList));
            this.storeTodo("CurrentTodoNr", id.toString());
        }
        else{alert("The todo needs a description!!");}
    };

    //This function removes the finished to-do from the AsyncStorage, and replaces it with "done" at the start of the id.
    handleFinishedTodo(list){
        const id = list[0]; const text = list[1]; const date = list[2]; const done = list[3];
        const AsyncList = [text,date,done];

        AsyncStorage.removeItem(id.toString());

        this.storeTodo("done"+id.toString(), JSON.stringify(AsyncList));
        this.setState({finishedTodoList:list});
    }

    //This function is triggered when the user press the button "Show finished todos"
    //It updates the list being to shown to the user with the preferred list (todos or finished todos)
    handleShowFinishedTodos(){
        this.setState({showFinishedTodos:!this.state.showFinishedTodos});
        if(this.state.showFinishedTodos){
            this.updateShowList(this.state.finishedTodoList)
        }
        else{
            this.updateShowList(this.state.todoList)
        }
    }

    //updateSortedList immediately updates the list when the date is changed.
    //if a to-do with an older date than another to-do is changed, the to-do will move above
    updateSortedList(list){
        const updateList = this.state.todoList.slice();
        updateList.map((item) => {
            //checks if the to-do which is changed matches an todonr in the todolist
            if(item.todoNr === list[0]){
                //update the todoDate to the new one
                item.todoDate = list[2];
            }
        });

        //The list is being sat to both to-do- and showlist for immediate change on the screen.
        this.updateTodoList(updateList);
        this.updateShowList(updateList);
    }

    //deleteTodo deletes an to-do from the todolist.
    //The function gets an id as parameter and, creates a new temporarily list, and adds each element from the todolist except
    //the element with the given id
    deleteTodo = (id) => {
        const deleteList = [];
        for(let i = 0; i < this.state.todoList.length; i++){
            if(this.state.todoList[i].todoNr !== id){
                deleteList.push(this.state.todoList[i]);
            }
        }
        //The list is being sat to both to-do- and showlist for immediate change on the screen.
        this.updateTodoList(deleteList);
        this.updateShowList(deleteList);

        //The item is also removed from the AsyncStorage
        AsyncStorage.removeItem(id.toString());
    };

    //This function sorts the current list by date
    sortByDate(todoList){
        const todosWithDate = [];
        const todosWithoutDate = [];

        todoList.map((item) => {
            if(item.todoDate !== ""){
                todosWithDate.push(item);
            }
            else{
                todosWithoutDate.push(item);
            }
        });

        todosWithDate.sort(function(a,b){
            if(a.todoDate !== "" && b.todoDate !== ""){
                let c = new Date(a.todoDate);
                let d = new Date(b.todoDate);
                return c-d;
            }
        });
        return todosWithDate.concat(todosWithoutDate);
    };

    //This is the list being shown to the user, and this function updates the list.
    updateShowList(list){
        const sortedByDate = this.sortByDate(list);
        this.setState({showList:sortedByDate});
    }

    //Updates the todolist, so the user get immediate update.
    updateTodoList(list){
        const sortedByDate = this.sortByDate(list);
        this.setState({todoList:sortedByDate});
    }

    //Function to store values in the AsyncStorage
    storeTodo = async (id, data) => {
        try {
            await AsyncStorage.setItem(id, data);
        } catch (error) {
            throw error;
        }
    };

    render() {
        return (
            <View style={styles.container}>
                <View style={styles.toolbar}>
                    {/*Textinput that automatically gets focused when the user enters the todopage.*/}
                    {/*When text is written, the todoTextstate gets changed*/}
                    <TextInput
                        placeholder="write something here"
                        ref={this.myTextInput}
                        style={styles.textInput}
                        onChangeText={(text) => this.setState({todoText : text})}
                        value={this.state.todoText}
                    />
                    {/*Creates a to-do with the given text written in the textinput, and the current todonr*/}
                    <TouchableOpacity style={styles.toolbarAddBtn} onPress={() => this.addTodo(this.state.currTodoNr)}>
                        <Text style={styles.toolbarAddBtnText}>Add</Text>
                    </TouchableOpacity>
                </View>
                <View  style={styles.finishedTodosBtn}>
                    <Button
                        onPress={() => this.handleShowFinishedTodos()}
                        title={this.state.showFinishedTodos ? "Show finished todos" : "Show todos"}
                    />
                </View>
                <View style={{flex:1}}>
                    {/*Scrollview so the user could scroll if the amount of todos become large*/}
                    {/*Maps through the todolist (which is a prop from app.js), and creates a listitem for each to-do*/}
                    {<ScrollView>
                        {this.state.showList.map((element) =>
                        <ListItem handleFinishedTodo = {this.handleFinishedTodo} updateSortedList = {this.updateSortedList} deleteTodo = {this.deleteTodo} name = {element.todoText} key = {element.key} done = {element.done} date = {element.todoDate} todoNr = {element.todoNr}/>
                    )}
                    </ScrollView>}
                </View>
            </View>
        );
    }
};