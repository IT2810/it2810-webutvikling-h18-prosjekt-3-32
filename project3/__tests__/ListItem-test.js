import React from "react";
import TodoList from "../components/TodoList";
import ListItem from "../components/ListItem";

import ShallowRenderer from "react-test-renderer/shallow";
import renderer from "react-test-renderer";
import Moment from "moment";

//Snapshot testing of ListItem.js
test("ListItem renders correctly!", () => {
    const tree = renderer.create(<ListItem/>).toJSON();
    expect(tree).toMatchSnapshot();
});