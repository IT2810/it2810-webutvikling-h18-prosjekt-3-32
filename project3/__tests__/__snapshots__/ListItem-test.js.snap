// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem renders correctly! 1`] = `
Array [
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ededed",
        "flex": 1,
        "flexDirection": "row",
        "height": 70,
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
          "borderColor": "transparent",
          "borderRadius": 3,
          "borderWidth": 1,
          "margin": 5,
          "marginLeft": -3,
          "marginRight": 10,
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
            false,
            false,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          onPress={[Function]}
          style={
            Array [
              Object {
                "color": "#bfbfbf",
                "fontSize": 24,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          ï‚–
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "#43484d",
                  "fontWeight": "bold",
                  "marginLeft": 10,
                  "marginRight": 10,
                },
                undefined,
                undefined,
              ],
            ]
          }
        />
      </View>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#7f7f7f",
          "fontSize": 16,
          "marginLeft": -30,
          "maxWidth": "78%",
          "width": "78%",
        }
      }
    />
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "height": 30,
            "width": 30,
          }
        }
      />
    </View>
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ededed",
        "borderBottomWidth": 0.25,
        "flex": 1,
        "height": 18,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#7f7f7f",
          "fontSize": 12,
        }
      }
    />
  </View>,
  <View>
    <Modal
      animationType="slide"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      transparent={false}
      visible={false}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "flex-start",
              "marginBottom": 15,
              "marginLeft": 10,
              "opacity": 1,
            }
          }
        >
          <Image
            source={
              Object {
                "process": [Function],
              }
            }
            style={
              Object {
                "height": 30,
                "padding": 5,
                "width": 30,
              }
            }
          />
        </View>
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          style={
            Object {
              "backgroundColor": "#dddddd",
              "fontSize": 25,
              "padding": 20,
              "textAlign": "center",
              "width": "100%",
            }
          }
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ededed",
              "borderBottomWidth": 0.2,
              "flexDirection": "row",
              "height": "18%",
              "justifyContent": "center",
              "width": "100%",
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "width": 142,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "height": 40,
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "borderColor": "#aaa",
                      "borderWidth": 1,
                      "flex": 1,
                      "height": 40,
                      "justifyContent": "center",
                    },
                    undefined,
                    false,
                    false,
                  ]
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#c9c9c9",
                      },
                      undefined,
                    ]
                  }
                >
                  select date
                </Text>
              </View>
              <Image
                source={
                  Object {
                    "process": [Function],
                  }
                }
                style={
                  Array [
                    Object {
                      "height": 32,
                      "marginLeft": 5,
                      "marginRight": 5,
                      "width": 32,
                    },
                    undefined,
                  ]
                }
              />
              <Modal
                animationType="none"
                hardwareAccelerated={false}
                onRequestClose={[Function]}
                supportedOrientations={
                  Array [
                    "portrait",
                    "portrait-upside-down",
                    "landscape",
                    "landscape-left",
                    "landscape-right",
                  ]
                }
                transparent={true}
                visible={false}
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                >
                  <View
                    accessible={true}
                    isTVSelectable={true}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "alignItems": "flex-end",
                        "backgroundColor": "#00000077",
                        "flex": 1,
                        "flexDirection": "row",
                      }
                    }
                  >
                    <View
                      accessible={true}
                      isTVSelectable={true}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "backgroundColor": "#fff",
                            "height": 0,
                            "overflow": "hidden",
                          }
                        }
                      >
                        <View
                          pointerEvents="auto"
                        >
                          <View
                            style={
                              Array [
                                Object {
                                  "borderTopColor": "#ccc",
                                  "borderTopWidth": 1,
                                  "marginTop": 42,
                                },
                                undefined,
                              ]
                            }
                          >
                            <RCTDatePicker
                              date={1539707925674}
                              minimumDate={1539707925720}
                              mode="date"
                              onChange={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={
                                Object {
                                  "height": 216,
                                }
                              }
                              timeZoneOffsetInMinutes={null}
                            />
                          </View>
                        </View>
                        <View
                          accessible={true}
                          isTVSelectable={true}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "flexDirection": "row",
                                "height": 42,
                                "justifyContent": "center",
                                "paddingHorizontal": 20,
                                "position": "absolute",
                                "top": 0,
                              },
                              Object {
                                "left": 0,
                              },
                              undefined,
                            ]
                          }
                        >
                          <Text
                            accessible={true}
                            allowFontScaling={true}
                            ellipsizeMode="tail"
                            style={
                              Array [
                                Object {
                                  "color": "#46cf98",
                                  "fontSize": 16,
                                },
                                Object {
                                  "color": "#666",
                                },
                                undefined,
                              ]
                            }
                          >
                            Cancel
                          </Text>
                        </View>
                        <View
                          accessible={true}
                          isTVSelectable={true}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "flexDirection": "row",
                                "height": 42,
                                "justifyContent": "center",
                                "paddingHorizontal": 20,
                                "position": "absolute",
                                "top": 0,
                              },
                              Object {
                                "right": 0,
                              },
                              undefined,
                            ]
                          }
                        >
                          <Text
                            accessible={true}
                            allowFontScaling={true}
                            ellipsizeMode="tail"
                            style={
                              Array [
                                Object {
                                  "color": "#46cf98",
                                  "fontSize": 16,
                                },
                                undefined,
                              ]
                            }
                          >
                            Confirm
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </Modal>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ededed",
              "borderBottomWidth": 0.2,
              "flexDirection": "row",
              "height": "18%",
              "justifyContent": "center",
              "width": "100%",
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#3aadff",
                "height": "100%",
                "justifyContent": "center",
                "opacity": 1,
                "width": "100%",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#f7f7f7",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              Save
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ededed",
              "borderBottomWidth": 0.2,
              "flexDirection": "row",
              "height": "18%",
              "justifyContent": "center",
              "width": "100%",
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ff3d3d",
                "height": "100%",
                "justifyContent": "center",
                "opacity": 1,
                "width": "100%",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#f7f7f7",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              Delete
            </Text>
          </View>
        </View>
      </View>
    </Modal>
  </View>,
]
`;
